code/modules/clothing/masks/chemmask.dm

```diff
@@ -0,0 +1,335 @@
+//the time interval, the amount injected, and the reagent monitored by the threshold system, through verbs.
+
+//By default, the primary tank is connected to the mask, and the auxiliary beaker is disconnected. The user may toggle these
+//connections using verbs. When disconnected, the mask will not take any additional reagents from the primary tank and/or the
+//auxiliary beaker. Cycling these verbs will not force injections, the tank is still only capable of one injection per 97 seconds,
+//and the beaker has no need to be forced to inject since the user can alter its time interval directly.
+
+/obj/item/clothing/mask/chemmask
+	desc = "A rather sinister mask designed for connection to a chemical pack, providing the pack's safeties are disabled."
+	name = "chemical mask"
+	icon_state = "chemmask0"
+	flags = FPRINT | MASKINTERNALS
+	w_class = 2
+	var/power = 0
+	gas_transfer_coefficient = 0.01
+	permeability_coefficient = 0.01
+	var/tank_injection_rates = list(5,10,15,20,25,30,35,40,45,50)
+	var/tank_injection_rate = 10
+	var/tank_has_injected = 0
+	var/beaker_injection_methods = list("Time-based","Threshold-based")
+	var/beaker_injection_method = "Threshold-based"
+	var/injection_method_chosen = 0
+	var/beaker_time_intervals = list(1,5,10,15,20,25,30,40,50,60,70,80,90,97,100,120,180,240,300)
+	var/beaker_time_interval = 970
+	var/beaker_injection_rates = list(5,10,15,20,25,30,35,40,45,50)
+	var/beaker_injection_rate = 10
+	var/beaker_thresholds = list(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,30,40,50)
+	var/beaker_threshold = 10
+	var/beaker_threshold_reagents = list("creatine","blood","anti_toxin","bicaridine","kelotane","dermaline","dexalinp","iron","hyperzine","doctorsdelight","inaprovaline","space_drugs","tramadol","oxycodone","dexalin","leporazine","tricordrazine","hyronalin","alkysine","imidazoline","peridaxon","peptobismol")
+	var/beaker_threshold_reagent = "creatine"
+	var/empty = 0
+	var/beakerempty = 0
+	var/hasbeaker = 0
+	var/beakeractive = 0
+	var/tankactive = 1
+	species_fit = list("Vox")
+	body_parts_covered = HEAD|MOUTH
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/chempack.dmi', "right_hand" = 'icons/mob/in-hand/right/chempack.dmi')
+	origin_tech = "biotech=5;materials=5;engineering=5;syndicate=5;combat=5"
+
+/obj/item/clothing/mask/chemmask/New() //Doing this so that these verbs don't show up before there's actually a beaker in the pack.
+	..()
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_pack_injection
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_injection_method
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_threshold
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_time_interval
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_threshold_reagent
+	verbs -= /obj/item/clothing/mask/chemmask/verb/set_beaker_injection_rate

```

Comment:
Add an update_verbs proc that adds and removes verbs so you don't have duplicated code everywhere.  Use a list for each category, like `chem_mask_verbs_time` and `chem_mask_verbs_threshold`.  Then you can just add and remove the list rather than the individual verbs.